src/common_tmr.c

```diff
@@ -79,6 +79,10 @@ static void vtmr_reset_timer( unsigned vid )
   unsigned id = VTMR_GET_ID( vid );
 
   vtmr_reset_idx = ( s8 )id;
+  
+  // TH: Ensure that Interrupts are enabled before timer is reset, otherwise eLua will hang forever....
+  platform_cpu_set_global_interrupts(PLATFORM_CPU_ENABLE);
+  // End TH
```

Comment:
The usual pattern for interrupt manipulation (which restores the previous state of interrupts) is:

```
int oldstate = platform_cpu_set_global_interrupts(PLATFORM_CPU_ENABLE);
// do your stuff here
platform_cpu_set_global_interrupts(oldstate);
```